src/routes/+page.svelte

```diff
@@ -0,0 +1,241 @@
+<script lang="ts">
+	import * as d3 from 'd3';
+	import type { DateEntry } from '$lib/types';
+
+	export let data;
+	let entries = data?.entries;
+
+	$: timerange = all;
+
+	const lastWeek = new Date(new Date().getTime() - 7 * 24 * 60 * 60 * 1000);
+	const lastMonth = new Date(new Date().getTime() - 7 * 4 * 24 * 60 * 60 * 1000);
+	const lastSixMonths = new Date(new Date().getTime() - 7 * 4 * 24 * 60 * 60 * 1000 * 6);
+	const lastYear = new Date(new Date().getTime() - 365 * 24 * 60 * 60 * 1000);
+	const all = new Date('2022-12-31');
+
+	$: category = null;
+
+	const gap = 0;
+	const padding = 32;
+	$: innerWidth = 1300;
+	$: innerHeight = 600;
+	$: width = innerWidth / 1.05;
+	$: height = innerHeight / 1.75;
+
+	let tidy = entries
+		.map((d: DateEntry) => [
+			{
+				date: d.date,
+				status: 'Images published with alternative text',
+				value: d.images_published_with_alt_text
+			},
+			{
+				date: d.date,
+				status: 'Images published',
+				value: d.images_published
+			}
+		])
+		.flat();
+
+	$: tidy = entries
+		.map((d: DateEntry) => [
+			{
+				date: d.date,
+				status: 'Images published with alternative text',
+				value: category
+					? JSON.parse(d.category_data)[category]?.images_published_with_alt_text ?? 0
+					: d.images_published_with_alt_text
+			},
+			{
+				date: d.date,
+				status: 'Images published',
+				value: category
+					? JSON.parse(d.category_data)[category]?.images_published ?? 0
+					: d.images_published
+			}
+		])
+		.flat()
+		.filter((entry) => entry.date > timerange.toISOString().split('T')[0]);
+
+	$: index = d3.rollup(
+		tidy,
+		(v) => Object.fromEntries(v.map((o) => [o.status, o.value])),
+		(d) => d.date
+	);
+
+	$: x = d3.scaleUtc(d3.extent(tidy.map((d) => new Date(d.date))) as Iterable<Number>, [
+		padding,
+		width - padding
+	]);
+	$: y = d3.scaleLinear(d3.extent(tidy.map((d) => d.value)) as Iterable<Number>, [
+		height - padding,
+		padding
+	]);
+
+	let gx: SVGGElement;
+	let gy: SVGGElement;
+	$: d3.select(gx).call(d3.axisBottom(x));
+	$: d3.select(gy).call(d3.axisLeft(y));
+
+	function handleMouseOver(d: MouseEvent, data: { date: string; values: Record<string, number> }) {
+		d3.select('.tooltip')
+			.style('display', 'block')
+			.style('left', d.pageX + 'px')
+			.style('top', d.pageY - 200 + 'px')
+			.text(
+				`On ${data.date}, ${data.values['Images published with alternative text']} out of ${data.values['Images published']} images had alt text (${((data.values['Images published with alternative text'] / data.values['Images published']) * 100).toFixed(2)}%)`
+			);
+
+		d3.selectAll('.stacked-bar').style('opacity', '0.25');
+		d.target!.style.opacity = 1;
+	}
+
+	function handleMouseOut(d: MouseEvent) {
+		if (d.toElement.nodeName !== 'rect') {
+			d3.select('.tooltip').style('display', 'none');
+			d3.selectAll('.stacked-bar').style('opacity', 1);
+		}
```

Comment:
I think you can remove this guard: 

```suggestion
	d3.select('.tooltip').style('display', 'none');
	d3.selectAll('.stacked-bar').style('opacity', 1);
```